lib/streamlit/watcher/util.py

```diff
@@ -30,28 +31,21 @@
 _RETRY_WAIT_SECS = 0.1
 
 
-def calc_md5_with_blocking_retries(file_path):
-    """Calculate the MD5 checksum of a given file_path
+def calc_md5_with_blocking_retries(path: str) -> str:
+    """Calculate the MD5 checksum of a given path.
+
+    For a file, this means calculating the md5 of the file's contents. For a
+    directory, we concatenate the directory's path with the names of all the
+    files in it and calculate the md5 of that.
 
     IMPORTANT: This method calls time.sleep(), which blocks execution. So you
     should only use this outside the main thread.
-
-    Parameters
-    ----------
-    file_path : str
-        The path of the file to check.
-
-    Returns
-    -------
-    str
-        The MD5 checksum.
-
     """
 
-    if os.path.isdir(file_path):
-        content = file_path.encode("UTF8")
+    if os.path.isdir(path):
```

Comment:
not necessary, but as a part of a migration to the modern Path module `[Path.is_dir()](https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_dir)` can be used here

Reply:
Will look into this one later. The tricky piece is that `Path.is_dir` is a method on `Path` objects, and `path` is passed into this helper function as a string.

I think we'll want to do the conversion from path string -> `Path` object earlier than in this helper function, but that will take a decent amount of work.